### ISSUE TITLE
Group set in editor does not follow instantiated children of Scene.

### ISSUE THREAD

**Issue by luiscesjr** [godotengine/godot#68370]
### Godot version

v4.0.beta1.official [20d667284]

### System information

Windows 11

### Issue description

(I'm sorry if the instance being gramatically wrong used here causes any trouble)

When instancing a Scene, the instantiated children do not inherit it's parent's groups, if set a group in the editor. A workaround to make sure the children also be a part of such group is explicitly writing:

`add_to_group("group_name")`

I'm not using the latest godot 4 beta, but I'm assuming it's the same across the other versions since I did not find a similar issue reported around here.

### Steps to reproduce

Create a Scene, import etc;
Give it a group using the editor like so:

![image](https://user-images.githubusercontent.com/7636427/200314349-9b3f648e-80d8-4078-8c09-871550a855b5.png)

Instance it around;
It's children/copies won't inherit the same group

**Comment by Zireael07** [comment]
Are they children OR copies? That's an important distinction. Do you duplicate (Ctrl+D) or just instantiate or just copy?

Children aren't meant to have the group.

**Comment by clayjohn** [comment]
I'm not sure if this is a bug or not. If I recall correctly, a group is specific to a SceneTree, so I wouldn't expect the new instances of the scene to be added to the group automatically.

**Comment by luiscesjr** [comment]
Not sure, maybe it was intended not to be added to the groups via editor? But I would think it's a bug, makes no sense.

It sure is misleading,  imagine spawning hundreds of enemies and not understanding why they do not function properly? 

So if a simple add_to_group solves the issue, I think that both cases should function the same, no?

**Comment by Maran23** [comment]
I'm also a bit confused. 
Let's say I have a scene called `Enemy` with the group `enemy.`
When I now create instances of `Enemy` via `instantiate()`, they have the group `enemy`.
This should work.

**Comment by luiscesjr** [comment]
> I'm also a bit confused. Let's say I have a scene called `Enemy` with the group `enemy.` When I now create instances of `Enemy` via `instantiate()`, they have the group `enemy`. This should work.

Yup, I also think it should, I don't know how it is on the newer betas, hadn't tested this in a while, I'm hard setting them in code now to be sure. Leads to some confused bug hunting haha.

Bear in mind this happens only when setting the group via editor.

**Comment by Botched135** [comment]
According to the step-by-step guide, this is the suggested way to call queue_free on a bunch of enemies:
https://docs.godotengine.org/en/stable/getting_started/first_2d_game/06.heads_up_display.html#removing-old-creeps


So it only make sense that they would be added during instantiation
